db/fixtures/ae_datastore/ManageIQ/System/Event/EmsEvent/RHEVM.class/user_remove_vm_finished.yaml

```diff
@@ -9,8 +9,8 @@ object:
     description: 
   fields:
   - rel4:
-      value: "/System/event_handlers/src_vm_disconnect_storage"
-  - rel5:
       value: "/System/event_handlers/event_action_policy?target=src_vm&policy_event=vm_unregister&param="
+  - rel5:
+      value: "/System/event_handlers/event_action_refresh_sync?target=src_vm"
```

Comment:
Well these things are hard to reproduce. It really depends on number of workers and the amount of work they are doing. Then you can see refresh always finished first, because the workers are processing disconnect_storage are just way more busy, or a part of the disconnect_storage runs before the refresh and part after.

So I would expect that sync refresh (or async with wait) should fix this. Then doing the refresh from the storage side should catch the rest (like event doesn't come or fails)

The main problem here is that Disk is nested under a VM, so in this case there needs to be some sort of cascade delete (cascade disconnect) right when we delete VM. (still there is a probability to get orphans when it's not DB level cascade delete)

And if the object lives on it's own (not a Vm's local disk), it should be really refreshed elsewhere, like as part of storage. I think we have this nicely done for cloud_volume, where disk is either local or a link to a cloud_volume. Storing Disk from 2 places, while not breaking other providers will be a bit more challenging, but not impossible. :-)